test/sdks-e2e/sdks_test.go

```diff
@@ -48,7 +51,45 @@ func (s *SDKsSuite) TestUDFunctionAndSink() {
 		SendMessageTo(pipelineName, "in", *NewRequestBuilder().WithBody([]byte("hello")).Build())
 
 	w.Expect().VertexPodLogContains("python-udsink", "hello", PodLogCheckOptionWithContainer("udsink"), PodLogCheckOptionWithCount(3)).
-		VertexPodLogContains("go-udsink", "hello", PodLogCheckOptionWithContainer("udsink"), PodLogCheckOptionWithCount(3))
+		VertexPodLogContains("go-udsink", "hello", PodLogCheckOptionWithContainer("udsink"), PodLogCheckOptionWithCount(3)).
+		VertexPodLogContains("java-udsink", "hello", PodLogCheckOptionWithContainer("udsink"), PodLogCheckOptionWithCount(3))
+}
+
+func (s *SDKsSuite) TestReduceSDK() {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+	w := s.Given().Pipeline("@testdata/simple-keyed-reduce-pipeline.yaml").
+		When().
+		CreatePipelineAndWait()
+	defer w.DeletePipelineAndWait()
+	pipelineName := "even-odd-sum"
+
+	// wait for all the pods to come up
+	w.Expect().VertexPodsRunning()
+
+	done := make(chan struct{})
```

Comment:
I guess this is not needed? Line 60 will take effect.

Reply:
yeah its not needed

Reply:
No, we need this, I want to stop publishing the messages even before deleting the pipeline.
I forgot the reason for adding it, now when I removed it the tests failed

Reply:
Oh... but you can simply call `cancel()`. Anyways, you make the call.

Reply:
Or switch the order of defers.

Reply:
I just wanted to stop publishing once I see the output in logs, so didn't want to use the context for that, so that the context can be shared with other methods and also we can add more assert statements in future.